### PR TITLE
Fix wrong memcached addresses

### PR DESCRIPTION
**Pull request by Skaronator**
**What this PR does**:
This fixes the following Error I got on my store-gateway with the latest helm chart (0.6.0)
```
name=metadata-cache msg="failed to resolve addresses for memcached" addresses=dns+cortex-memcached-blocks-metadata.cortex.svc:11211 err="lookup IP addresses \"cortex-memcached-blocks-metadata.cortex.svc\": i/o timeout"
```

The issue is that `cortex-memcached-blocks-metadata.cortex.svc` doesn't work. Either just use the service name (when its in the same namespace) so `cortex-memcached-blocks-metadata` or use the full svc dns record which would be `cortex-memcached-blocks-metadata.cortex.svc.cluster.local`.

```bash
/ # nslookup cortex-memcached-blocks.cortex.svc
Server:		172.20.0.10
Address:	172.20.0.10:53

** server can't find cortex-memcached-blocks.cortex.svc: NXDOMAIN

** server can't find cortex-memcached-blocks.cortex.svc: NXDOMAIN
```
```
/ # nslookup cortex-memcached-blocks.cortex.svc.cluster.local
Server:		172.20.0.10
Address:	172.20.0.10:53

*** Can't find cortex-memcached-blocks.cortex.svc.cluster.local: No answer

Name:	cortex-memcached-blocks.cortex.svc.cluster.local
Address: 10.5.12.79
Name:	cortex-memcached-blocks.cortex.svc.cluster.local
Address: 10.5.26.70

```

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: templates/_helpers.tpl [hunk 1]
```diff
@@ -87,20 +87,20 @@ Create configuration parameters for memcached configuration
...
-- "-blocks-storage.bucket-store.metadata-cache.memcached.addresses=dns+{{ .Release.Name }}-memcached-blocks-metadata.{{ .Release.Namespace }}.svc:11211"
+- "-blocks-storage.bucket-store.metadata-cache.memcached.addresses=dns+{{ .Release.Name }}-memcached-blocks-metadata.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}:11211"
 {{- end -}}
 {{- if and (ne .Values.config.storage.engine "blocks") .Values.memcached.enabled }}
-- -store.chunks-cache.memcached.addresses=dns+{{ .Release.Name }}-memcached.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}:11211
+- "-store.chunks-cache.memcached.addresses=dns+{{ .Release.Name }}-memcached.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}:11211"
 {{- end -}}
 {{- if and (ne .Values.config.storage.engine "blocks") (index .Values "memcached-index-read" "enabled") }}
-- -store.index-cache-read.memcached.addresses=dns+{{ .Release.Name }}-memcached-index-read.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}:11211
+- "-store.index-cache-read.memcached.addresses=dns+{{ .Release.Name }}-memcached-index-read.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}:11211"
```
Comment (ShuzZzle) [inline templates/_helpers.tpl:100]: I think the Quotes are wrong here, but I didn't check

Comment (ShuzZzle) [reply templates/_helpers.tpl:100]: might be the reason why the build is failing

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## master / unreleased
 
+* [BUGFIX] Fix wrong memcached address for block storage
```
Comment (ShuzZzle) [inline CHANGELOG.md:5]: add pr number pls :)

**Comment by ShuzZzle** [comment]
> The issue is that `cortex-memcached-blocks-metadata.cortex.svc` doesn't work. Either just use the service name (when its in the same namespace) so `cortex-memcached-blocks-metadata` or use the full svc dns record which would be `cortex-memcached-blocks-metadata.cortex.svc.cluster.local`.

That entirely depends on how you configured your dns search domain. For example the config works fine in my cluster at the moment.

```
k exec -it dnsutils -- nslookup cortex-memcached-blocks-metadata.cortex.svc                                          
Server:		10.43.0.10
Address:	10.43.0.10#53

Name:	cortex-memcached-blocks-metadata.cortex.svc.cluster.local
Address: 10.42.8.156
Name:	cortex-memcached-blocks-metadata.cortex.svc.cluster.local
Address: 10.42.6.66

```

However I am a Fan of explicit rather than implicit, therefore i am +1 for this change. We should just specify the entire FQDN

File: CHANGELOG.md (hunk 2 above)
Comment (ShuzZzle) [reply CHANGELOG.md:5]: maybe replace BUGFIX with ENHANCEMENT

File: templates/_helpers.tpl (hunk 1 above)
Comment (Skaronator) [reply templates/_helpers.tpl:100]: Mh build fails due to Nodes not getting ready. Hopefully my new build just works again.

Comment (ShuzZzle) [reply templates/_helpers.tpl:100]: yep the kind cluster is failing, but why lol

**Comment by ShuzZzle** [comment]
I "fixxed" the pipeline by upgrading kind cluster (necessary anyway) in #177. Seems to be back to working order now and the nodes are actually ready this time.

**Review by ShuzZzle — APPROVED** [review]

**Pull request merged**